src/resin-mixpanel-client.js

```diff
@@ -124,6 +124,34 @@ module.exports = function(token, options) {
 			} else {
 				throw new Error('(Resin Mixpanel Client) function identify is only available for the browser')
 			}
+		},
+		associateDeviceIds: function(deviceIds) {
+			if (!isBrowser) {
+				throw new Error('(Resin Mixpanel Client) function associateDeviceIds is only available for the browser');
+			}
+			var userName = mixpanel.get_property('$user_id');
+			var distinctId = mixpanel.get_property('distinct_id');
+			if (!userName) {
+				throw new Error('(Resin Mixpanel Client) mixpanel does not have correctly identified user info');
+			}
+			var originalDeviceId = mixpanel.get_property('$device_id');
+
+			// Send alias events to combine all input devices under a single user name.
+			for (var i = 0; i < deviceIds.length; i++) {
+				var deviceId = deviceIds[i];
+				mixpanel.register({
```

Comment:
Does Mixpanel send an explicit event here? Like `$create_alias`. 

Reading the docs, I understand that `register` sets a super property in Mixpanel world. Then the code overwrites with all the values and sets it back to the original one. If all we need is the `$create_alias` event for Amplitude, why do we need `register`?

Reply:
No, this call only alters the internal state on the client without any API calls. We use `register` here to override distinct_id and $device_id fields and ensure that `$create_alias` API call triggered by the next line provides the data we need on the analytics proxy side.

Reply:
Mixpanel docs say that `register` makes a call to set super properties:
https://developer.mixpanel.com/docs/javascript-full-api-reference#section-mixpanel-register

Then as we discussed in our call today, I'm surprised that Mixpanel doesn't show such super property or link users in your examples. Maybe Mixpanel rejects this call due to usage of the default keywords like `distinct_id` and `$device_id`. That would explain how `alias` wouldn't work.

Reply:
I don't see any wording in the docs that says it's making an API call. And it does not during tests. And the source code shows it does not. :)
https://github.com/mixpanel/mixpanel-js/blob/master/src/mixpanel-core.js#L450
It only updates values in memory, cookie and/or local storage.

But set properties are used in track/identify calls later.

Reply:
Ah Mixpanel docs confused me. Still, I hold my theory that the reason we don't see these as super properties is because of using the Mixpanel reserved keywords. Then I understand that our intention is not about setting properties here but only to store them for future merging.

Sorry for the noise : )